cmd/marker/main.go

```diff
@@ -16,27 +16,43 @@ package main
 
 import (
 	"flag"
+	"fmt"
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/kubevirt/bridge-marker/pkg/cache"
 	"github.com/kubevirt/bridge-marker/pkg/marker"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 func main() {
 	nodeName := flag.String("node-name", "", "name of kubernetes node")
-	pollInterval := flag.Int("poll-interval", 10, "interval between updates in seconds, 10 by default")
+	const defaultUpdateInterval = 60 * time.Second
+	updateInterval := flag.Int("update-interval", int(defaultUpdateInterval.Seconds()), fmt.Sprintf("interval between updates in seconds, %d by default", defaultUpdateInterval))
+
+	const defaultReconcileInterval = 10 * time.Minute
+	reconcileInterval := flag.Int("reconcile-interval", int(defaultReconcileInterval.Minutes()), fmt.Sprintf("interval between node bridges reconcile in minutes, %d by default", defaultReconcileInterval))
 
 	flag.Parse()
 
 	if *nodeName == "" {
 		glog.Fatal("node-name must be set")
 	}
 
-	for {
-		err := marker.Update(*nodeName)
+	cache := cache.Cache{}
+	wait.JitterUntil(func() {
+		jitteredReconcileInterval := wait.Jitter(time.Duration(*reconcileInterval)*time.Minute, 1.2)
+		if time.Now().Sub(cache.LastRefreshTime()) >= jitteredReconcileInterval {
+			reportedBridges, err := marker.GetReportedResources(*nodeName)
+			if err != nil {
+				glog.Errorf("GetReportedResources failed: %v", err)
+			}
+			cache.Refresh(reportedBridges)
```

Comment:
We should add a log here to log a warning in case resources reported on the Node are not the same as kept in the cache. We don't expect it to happen, if it does, something may be wrong with the cluster and we want it to be visible in logs.

Reply:
Done.